library/src/blas1/rocblas_dot.hpp

```diff
@@ -0,0 +1,140 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#include "handle.h"
+#include "logging.h"
+#include "reduction_strided_batched.h"
+#include "rocblas.h"
+#include "utility.h"
+
+template <rocblas_int NB, bool CONJ, typename T, typename U, typename V = T>
+__global__ void dot_kernel_part1(rocblas_int    n,
+                                 const U        xa,
+                                 rocblas_int    offsetx,
+                                 rocblas_int    incx,
+                                 rocblas_stride stridex,
+                                 const U        ya,
+                                 rocblas_int    offsety,
+                                 rocblas_int    incy,
+                                 rocblas_int    stridey,
+                                 V*             workspace)
+{
+    ptrdiff_t tx  = hipThreadIdx_x;
+    ptrdiff_t tid = hipBlockIdx_x * hipBlockDim_x + tx;
+
+    __shared__ V tmp[NB];
+    const T*     x;
+    const T*     y;
+
+    // bound
+    if(tid < n)
+    {
+        x = load_ptr_batch(xa, hipBlockIdx_y, offsetx, stridex);
+        y = load_ptr_batch(ya, hipBlockIdx_y, offsety, stridey);
+        if(incx < 0)
+            x -= ptrdiff_t(incx) * (n - 1);
+        if(incy < 0)
+            y -= ptrdiff_t(incy) * (n - 1);
+        tmp[tx] = V(y[tid * incy]) * V(CONJ ? conj(x[tid * incx]) : x[tid * incx]);
```

Comment:
This is too expensive to have in device code. This will produce a regression in performance for the current non-batched `dot`.